jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/HttpChannelState.java

```diff
@@ -837,6 +837,8 @@ protected boolean onError(Throwable th)
                 {
                     if (_asyncListeners == null || _asyncListeners.isEmpty())
                     {
+//                      Case 3
+//                        return false;
```

Comment:
@gregw said:  _There is nothing explicit in the servlet spec saying that calling startAsync() should change the way exceptions thrown from a service method are handled. However, since the servlet spec allows exceptions to be matched to error pages, which are served with an ERROR dispatch, it would be nonsensical to have a synchronous ERROR dispatch happening whilst an async cycle was in progress. So if we are ASYNC, we should definitely not call sendError - EVEN IF there are no listeners registered._

_So I think this change is good._

Reply:
Hmm yes it is more nuanced.   The application should be able to call sendError and then complete.
But we should not call sendError on behalf of the application.